internal/controller/storageclient_controller.go

```diff
@@ -791,3 +833,15 @@ func (r *storageClientReconcile) getOperatorVersion() (string, error) {
 	}
 	return csv.Spec.Version.String(), nil
 }
+
+func ObjectsIterator[T client.Object](slices ...[]T) iter.Seq[T] {
+	return func(yield func(T) bool) {
+		for _, slice := range slices {
+			for idx := range slice {
+				if !yield(slice[idx]) {
+					return
+				}
+			}
+		}
+	}
+}
```

Comment:
This complexity is definitely not needed to accomplish what is expected from this PR. Please remove